server-spi/src/main/java/org/keycloak/storage/SearchableModelField.java

```diff
@@ -40,31 +40,6 @@ public String getName() {
         return fieldClass;
     }
 
-    @Override
-    public int hashCode() {
-        int hash = 5;
-        hash = 83 * hash + Objects.hashCode(this.name);
-        return hash;
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-        if (this == obj) {
-            return true;
-        }
-        if (obj == null) {
-            return false;
-        }
-        if (getClass() != obj.getClass()) {
-            return false;
-        }
-        final SearchableModelField<?> other = (SearchableModelField<?>) obj;
-        if ( ! Objects.equals(this.name, other.name)) {
-            return false;
-        }
-        return true;
-    }
-
```

Comment:
Why is this removed?

Reply:
Because of [this](https://github.com/mhajas/keycloak/blob/8feef482dc8400ba37aa128d101923d7b435eda0/model/map/src/main/java/org/keycloak/models/map/storage/MapFieldPredicates.java#L93): 

```java
private static final Map<SearchableModelField<?>, Comparator<?>> COMPARATORS = new HashMap<>();
```
and 
we have more `SearchableModelFields` from more models with the same name (for example CLIENT_ID).

Reply:
It seems we should come up with a better `equals` then :)

Reply:
Do we need a custom `equals`? Do we plan to, for example, send a `SearchableField` instance to another Keycloak node? If not we can have the default `equals` because all fields are static objects so within one VM a searchableField will be always equal only to itself.

Reply:
Thanks, you are right. In case of e.g. Infinispan, the infinispan representation of predicate would be generated from this representation within the same JVM, and only that representation (not the original model fields) would be sent over the wire.